tests/transforms/test_compile.py

```diff
@@ -0,0 +1,406 @@
+
+        qfunc = build_qfunc(wires)
+        dev = qml.device("default.qubit", wires=wires)
+
+        qnode = qml.QNode(qfunc, dev)
+
+        transformed_qfunc = compile(pipeline=[])(qfunc)
+        transformed_qnode = qml.QNode(transformed_qfunc, dev)
+
+        original_result = qnode(0.3, 0.4, 0.5)
+        transformed_result = transformed_qnode(0.3, 0.4, 0.5)
+        assert np.allclose(original_result, transformed_result)
+
+        names_expected = [op.name for op in qnode.qtape.operations]
+        wires_expected = [op.wires for op in qnode.qtape.operations]
+
+        compare_operation_lists(transformed_qnode.qtape.operations, names_expected, wires_expected)
+
+    @pytest.mark.parametrize(("wires"), [["a", "b", "c"], [0, 1, 2], [3, 1, 2], [0, "a", 4]])
+    def test_default_pipeline(self, wires):
+        """Test that the default pipeline returns the correct results."""
+
+        qfunc = build_qfunc(wires)
+        dev = qml.device("default.qubit", wires=Wires(wires))
+
+        qnode = qml.QNode(qfunc, dev)
+
+        transformed_qfunc = compile()(qfunc)
+        transformed_qnode = qml.QNode(transformed_qfunc, dev)
+
+        original_result = qnode(0.3, 0.4, 0.5)
+        transformed_result = transformed_qnode(0.3, 0.4, 0.5)
+        assert np.allclose(original_result, transformed_result)
+
+        names_expected = ["Hadamard", "CNOT", "RX", "CY", "PauliY"]
+        wires_expected = [
+            Wires(wires[0]),
+            Wires([wires[2], wires[1]]),
+            Wires(wires[0]),
+            Wires([wires[1], wires[2]]),
+            Wires(wires[2]),
+        ]
+
+        compare_operation_lists(transformed_qnode.qtape.operations, names_expected, wires_expected)
+
+    @pytest.mark.parametrize(("wires"), [["a", "b", "c"], [0, 1, 2], [3, 1, 2], [0, "a", 4]])
+    def test_pipeline_with_non_default_arguments(self, wires):
+        """Test that using non-default arguments returns the correct results."""
+
+        qfunc = build_qfunc(wires)
+        dev = qml.device("default.qubit", wires=Wires(wires))
+
+        qnode = qml.QNode(qfunc, dev)
+
+        pipeline = [
+            commute_controlled(direction="left"),
+            cancel_inverses,
+            merge_rotations(atol=1e-6),
+        ]
+
+        transformed_qfunc = compile(pipeline=pipeline)(qfunc)
+        transformed_qnode = qml.QNode(transformed_qfunc, dev)
+
+        original_result = qnode(0.3, 0.4, 0.5)
+        transformed_result = transformed_qnode(0.3, 0.4, 0.5)
+        assert np.allclose(original_result, transformed_result)
+
+        names_expected = ["Hadamard", "CNOT", "RX", "PauliY", "CY"]
+        wires_expected = [
+            Wires(wires[0]),
+            Wires([wires[2], wires[1]]),
+            Wires(wires[0]),
+            Wires(wires[2]),
+            Wires([wires[1], wires[2]]),
+        ]
+
+        compare_operation_lists(transformed_qnode.qtape.operations, names_expected, wires_expected)
```

Comment:
Similar thought here; it should be sufficient to have a test that ensures `merge_rotations` and `commute_controlled` simply are called with the correct arguments; testing the output is much more of a full-pipeline integration test, testing not just compile but also the three optimization transforms.

Reply:
Do you think it's worth determining how to test the arguments, or fine to just leave as testing the number of calls to each transform?

Reply:
I think that's fine :+1: